### ISSUE TITLE
Faulty d.ts in version 2.3.2

### ISSUE THREAD

**Issue by rogiia** [RasCarlito/axios-cache-adapter#96]
Since I updated axios-cache-adapter to last version 2.3.2, whenever I build my project with `tsc` the following error pops up:

`node_modules/axios-cache-adapter/axios-cache-adapter.d.ts:104:2 - error TS7008: Member 'store' implicitly has an 'any' type.`

`104  store;`
`    ~~~~~`

`Found 1 error.`

Looking at the d.ts file, the interface ISetupCache is indeed missing a type for the store property.

**Comment by RasCarlito** [comment]
Hey @rogiia !

Thanks for reporting this. I don't know much about typescript.

The `store` could be an instance of `localforage` which has [its own typings](https://github.com/localForage/localForage/blob/master/typings/localforage.d.ts) but it could also be an instance of internal [`MemoryStore` class](https://github.com/RasCarlito/axios-cache-adapter/blob/master/src/memory.js).

I don't know if I could express that the `ISetupCache.store` is one of those or if I should just simply express that it is an `object`.

What do you think?

Cheers 🍻

**Comment by rogiia** [comment]
Is the `store` property expected to only contain `LocalForage` instances? If that is right, then yes, `store: LocalForage` ought to do it. On the other hand, if this property is expected to also accept other types of instances, then we will need to find out which ones.

**Comment by atombrenner** [comment]
I have a similiar issue:
```node_modules/axios-cache-adapter/axios-cache-adapter.d.ts(104,2): error TS7008: Member 'store' implicitly has an 'any' type.```

Setting store to explicitly having the `any` type would fix this:
```
export interface ISetupCache
{
	adapter: AxiosAdapter;
	config: IAxiosCacheAdapterOptions;
	store: any; // explicit any type
}
```

**Comment by rogiia** [comment]
The use of `any` in Typescript in highly unrecommended. If the type can be any type that the user desires, I would use generics.

The interfaces would be declared like that:
```
export interface IAxiosCacheAdapterOptions<T>
...
store?: T;
```

```
export interface ISetupCache<T>
{
	adapter: AxiosAdapter;
	config: IAxiosCacheAdapterOptions;
	store<T>;
}
```
and finally the setupCache method would be:
`export declare function setupCache<T = MemoryStorage>(options: IAxiosCacheAdapterOptions<T>) : ISetupCache<T>;`

I have not tested this code but I feel like it should be the correct solution. What do you think?

**Comment by RasCarlito** [comment]
Fixed in [v2.3.3](https://github.com/RasCarlito/axios-cache-adapter/releases/tag/2.3.3)
